### PR TITLE
[TensorExpr] Add a skeleton for a registry of NNC lowerings.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65555
* #65859
* #65852
* #65843
* #65554
* #65553
* #65552
* #65551
* __->__ #65550
* #65549

This PR adds the source files and the class for the registry, subsequent
PRs actually port existing lowerings to this mechanism.

Differential Revision: [D31148922](https://our.internmc.facebook.com/intern/diff/D31148922)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

File: torch/csrc/jit/tensorexpr/kernel.cpp [hunk 1]
```diff
@@ -2455,6 +2455,13 @@ c10::optional<ScalarType> findDtypeForValue(const torch::jit::Value* v) {
   return c10::nullopt;
 }
 
+NNCLoweringFunction getStandardLoweringFor(const std::string& op) {
+  const auto& lowerings = getNNCLoweringRegistry();
+  if (lowerings.count(op))
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/kernel.cpp:2460]: nit: add braces '{...}', just to keep the coding style consistent.

**Review by navahgar — APPROVED** [review]

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31148922).

**Pull request closed without merging**